Updated to Discord.js v14

eirikhanasand opened the pull request:
- Closes #21 
- Closes #23 
- Updated to discord.js v14
- Added varius refactors necesarry by v14 breaking changes
- Fixed issues in discord.js-docs library by submitting a pr to the actual library instead of a non-maintained fork

eirikhanasand committed:
Updated dependencies, ecmascript target and added server alias for guild

eirikhanasand committed:
v14, top level error handlers, fixed djs and mdn docs

eirikhanasand committed:
Removed unused deps

eirikhanasand committed:
Started using linting, was defined but unused / no npm script for it

eirikhanasand committed:
Fixed autocomplete not working issues

eirikhanasand committed:
Replaced tabs with spaces

eirikhanasand committed:
Removed linting changes

eirikhanasand committed:
Fixed typos

eirikhanasand committed:
Removed excess code

khinshankhan commented on package.json:
```diff
@@ -15,29 +17,22 @@
-    "flexsearch": "^0.7.21",
-    "glob": "^7.2.0",
-    "node-fetch": "^2.6.6",
-    "pm2": "^5.1.2"
```
Why was pm2 removed?

eirikhanasand replied:
Because it was installed but there is no reference to it in the code, nor is it referenced by any other dependency neither top level nor sublevel nor any script nor the dockerfile. In other words, I found no indication that it was being used. Saw no reason to include it when it isnt used.

eirikhanasand committed:
Updated discord.js-docs and constantized default query

eirikhanasand commented:
Added a commit for discord.js-docs v 0.4.3 which was published earlier today, where an issue in that library that caused a top level error was resolved (see https://github.com/TeeSeal/discord.js-docs/pull/30). I also changed the default query "Client" to "a", and moved this to constants.ts and named it `DEFAULT_SEARCH_QUERY` to make it more explicit what the role of this variable is. Previous to this PR there was no default query, which resulted in no results appearing before you started typing. I feel like an alphabetically ordered list is better than just "Client" appearing, which was why I changed it to just "a".

eirikhanasand committed:
Moved DJSAutocomplete to its respective file and added the same default search query to mdn

eirikhanasand commented:
Added DEFAULT_SEARCH_QUERY for mdn docs as well (this was already "a", but now refers to the same constant instead of just a inline char. Also moved the DJSAutocomplete from `InteractionCreateHandler.ts` -> `commands/docs/djs.ts` to ensure both files follow the same structure.
